From c15962741d01915f74742452facb95d3f7518dc9 Mon Sep 17 00:00:00 2001
From: Yu Yin <xwafish@gmail.com>
Date: Tue, 17 Aug 2021 10:24:20 +0800
Subject: [PATCH] [mips64][riscv64] Export get_pc and get_register

Fix build failed with:
is_component_build = true
use_goma = true

Change-Id: Ia06175c091e94e36aa71c134b056b4d6b88e5c96
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3098826
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Reviewed-by: Zhao Jiazhong <zhaojiazhong-hf@loongson.cn>
Commit-Queue: Zhao Jiazhong <zhaojiazhong-hf@loongson.cn>
Cr-Commit-Position: refs/heads/master@{#76331}
---
 src/execution/mips64/simulator-mips64.h   | 4 ++--
 src/execution/riscv64/simulator-riscv64.h | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/execution/mips64/simulator-mips64.h b/src/execution/mips64/simulator-mips64.h
index ce3f06f2ed..69e8094174 100644
--- a/src/execution/mips64/simulator-mips64.h
+++ b/src/execution/mips64/simulator-mips64.h
@@ -243,7 +243,7 @@ class Simulator : public SimulatorBase {
   void set_register(int reg, int64_t value);
   void set_register_word(int reg, int32_t value);
   void set_dw_register(int dreg, const int* dbl);
-  int64_t get_register(int reg) const;
+  V8_EXPORT_PRIVATE int64_t get_register(int reg) const;
   double get_double_from_register_pair(int reg);
   // Same for FPURegisters.
   void set_fpu_register(int fpureg, int64_t value);
@@ -291,7 +291,7 @@ class Simulator : public SimulatorBase {
   unsigned int get_msacsr_rounding_mode();
   // Special case of set_register and get_register to access the raw PC value.
   void set_pc(int64_t value);
-  int64_t get_pc() const;
+  V8_EXPORT_PRIVATE int64_t get_pc() const;
 
   Address get_sp() const { return static_cast<Address>(get_register(sp)); }
 
diff --git a/src/execution/riscv64/simulator-riscv64.h b/src/execution/riscv64/simulator-riscv64.h
index 2fa40cea4e..50f105807a 100644
--- a/src/execution/riscv64/simulator-riscv64.h
+++ b/src/execution/riscv64/simulator-riscv64.h
@@ -312,7 +312,7 @@ class Simulator : public SimulatorBase {
   void set_register(int reg, int64_t value);
   void set_register_word(int reg, int32_t value);
   void set_dw_register(int dreg, const int* dbl);
-  int64_t get_register(int reg) const;
+  V8_EXPORT_PRIVATE int64_t get_register(int reg) const;
   double get_double_from_register_pair(int reg);
 
   // Same for FPURegisters.
@@ -354,7 +354,7 @@ class Simulator : public SimulatorBase {
 
   // Special case of set_register and get_register to access the raw PC value.
   void set_pc(int64_t value);
-  int64_t get_pc() const;
+  V8_EXPORT_PRIVATE int64_t get_pc() const;
 
   Address get_sp() const { return static_cast<Address>(get_register(sp)); }
 
-- 
2.35.1

